### DISCUSSION TITLE
Unknown terminal type when creating a release tag

### DISCUSSION THREAD

**Discussion by soywod** [magit/magit#4874] [Support]
Since the update of my system (I am running on NixOS), I cannot create release tag anymore:

```
Waiting for Emacs...
*ERROR*: Unknown terminal type
erreur : There was a problem with the editor '/nix/store/imy0i3xz2f8ixsv57lw5mf78p3q6c7i1-editor'.
```

It is the only command that does not work anymore, other commands spawning buffers like `commit` or even `tag create` work well. I checked a bit the source code and I am wondering why `magit-tag-create` uses `magit-run-git-with-editor` whereas `magit-tag-release` uses `magit-run-git-async`. I tried different debug options but I get nothing more. Let me know if you need any further information.

**Comment by tarsius** [comment]
I am guessing there is a package `editor` on NixOS and that it does not actually provide an `editor` executable but that installing it instead arranges for some editor to be called, either by setting the `EDITOR` variable in your shell's init file or by providing a wrapper script. It seems that wrapper script, or the executable it calls, expects a terminal. But that's all just speculation, I don't use NixOS. So you will have to figure out yourself what is going on here. Not installing the `editor` package might do the trick.

> I checked a bit the source code and I am wondering why `magit-tag-create` uses `magit-run-git-with-editor` whereas `magit-tag-release` uses `magit-run-git-async`.

The former calls `git tag` in a way that causes it to use the `$EDITOR`, so that you can write the message. The latter uses `-m "some generated message"`.

**Reply by soywod** [reply]
I initially opened an issue because `editor` is indeed a script generated by `NixOS` but it works on its own (and also works for ALL other commands based on `$EDITOR` like `commit` or `rebase`, and also works for ALL other programs based on `$EDITOR`). So my intuition tells me it comes either from emacs or magit.

> The former calls git tag in a way that causes it to use the $EDITOR, so that you can write the message. The latter uses -m "some generated message".

Which does not make sense, because `magit-tag-create` (which uses `$EDITOR`) works (the buffer spawns well and I can write my message), but `magit-tag-release` does not using `-m "message"` :thinking:

**Reply by tarsius** [reply]
You might want to try evaluating `(magit-run-git-async "tag" "-m" "foo" "bar")` directly to see whether that is really where the issue is triggered.

I've looked for relevant git variables. There is no `tag.alwaysEditMessage` variable or such. `core.editor` is interesting though:

```
      core.editor
           Commands such as commit and tag that let you edit messages by
           launching an editor use the value of this variable when it is set,
           and the environment variable GIT_EDITOR is not set. See git-var(1).
```

Sounds like `$GIT_EDITOR` > `core.editor` > `$EDITOR`.

I think we set `EDITOR` instead, so a user could mess that up by setting `core.editor`. Maybe installing the `editor` package does set that variable somehow.

**Reply by tarsius** [reply]
Na, Magit used `GIT_EDITOR`. (But not when using `magit-run-git-async`. That doesn't set either variable because when we use that, we don't expect `git` to use an editor.

**Comment by soywod** [comment]
> You might want to try evaluating `(magit-run-git-async "tag" "-m"
> "foo" "bar")` directly to see whether that is really where the issue
> is triggered.

Interesting, this works well! So the problem does not come from there.

**Reply by tarsius** [reply]
Try `toggle-debug-on-error` to narrow down the issue. Or if that does not help, edebug or debug statements in `magit-tag-release`.
